src/euphorie/content/surveygroup.py

```diff
@@ -134,6 +135,13 @@ def handleSurveyPublish(survey, event):
         return
     surveygroup = aq_parent(aq_inner(survey))
     surveygroup.published = survey.id
+    # Retract all other surveys if needed, to make sure they are in draft state.
+    for content in surveygroup.contentValues():
+        if content.portal_type != "euphorie.survey":
+            continue
+        if content.id == survey.id:
+            continue
+        api.content.transition(obj=content, to_state="draft")
```

Comment:
It will hardly take extra time.  This only gets the other surveys in this group and retract them if needed, without changing anything on the client side.  The client side stuff (adding/removing/updating items in the client folder in Plone and same in the Postgres database) is what takes up the bulk of the time, and that is not happening with these extra lines.

Reply:
I do not think the postgres DB is involved in this phase. I think most of the time is wasted reindexing.

Reply:
Ah, right, the Postgres DB probably only gets involved once someone starts actually filling in a survey on the client side.